application/comit_node/src/http_api/rfc003/handlers/post_swap.rs

```diff
@@ -0,0 +1,174 @@
+        match identities {
+            SwapRequestBodyIdentities::RefundAndRedeem { .. }
+            | SwapRequestBodyIdentities::OnlyRefund { .. }
+            | SwapRequestBodyIdentities::None {} => {
+                Err(HttpApiProblem::with_title_and_type_from_status(400))
+            }
+            SwapRequestBodyIdentities::OnlyRedeem {
+                beta_ledger_redeem_identity,
+            } => Ok(rfc003::alice::SwapRequestIdentities {
+                alpha_ledger_refund_identity: secret_source.new_secp256k1_refund(id),
+                beta_ledger_redeem_identity,
+            }),
+        }
+    }
+}
+
+impl FromSwapRequestBodyIdentities<Ethereum, Bitcoin>
+    for rfc003::alice::SwapRequestIdentities<Ethereum, Bitcoin>
+{
+    fn from_swap_request_body_identities(
+        identities: SwapRequestBodyIdentities<
+            ethereum_support::Address,
+            bitcoin_support::PubkeyHash,
+        >,
+        id: SwapId,
+        secret_source: &dyn SecretSource,
+    ) -> Result<Self, HttpApiProblem> {
+        match identities {
+            SwapRequestBodyIdentities::RefundAndRedeem { .. }
+            | SwapRequestBodyIdentities::OnlyRedeem { .. }
+            | SwapRequestBodyIdentities::None {} => {
+                Err(HttpApiProblem::with_title_and_type_from_status(400))
+            }
+            SwapRequestBodyIdentities::OnlyRefund {
+                alpha_ledger_refund_identity,
+            } => Ok(rfc003::alice::SwapRequestIdentities {
+                alpha_ledger_refund_identity,
+                beta_ledger_redeem_identity: secret_source.new_secp256k1_redeem(id),
+            }),
+        }
+    }
+}
+
+trait FromSwapRequestBody<AL: Ledger, BL: Ledger, AA: Asset, BA: Asset>
+where
+    Self: Sized,
+{
+    fn from_swap_request_body(
+        body: SwapRequestBody<AL, BL, AA, BA>,
+        id: SwapId,
+        secret_source: &dyn SecretSource,
+    ) -> Result<Self, HttpApiProblem>;
+}
+
+impl<AL: Ledger, BL: Ledger, AA: Asset, BA: Asset> FromSwapRequestBody<AL, BL, AA, BA>
+    for rfc003::alice::SwapRequest<AL, BL, AA, BA>
+where
+    SwapRequestIdentities<AL, BL>: FromSwapRequestBodyIdentities<AL, BL>,
+{
+    fn from_swap_request_body(
+        body: SwapRequestBody<AL, BL, AA, BA>,
+        id: SwapId,
+        secret_source: &dyn SecretSource,
+    ) -> Result<Self, HttpApiProblem> {
+        Ok(rfc003::alice::SwapRequest {
+            alpha_asset: body.alpha_asset,
+            beta_asset: body.beta_asset,
+            alpha_ledger: body.alpha_ledger,
+            beta_ledger: body.beta_ledger,
+            alpha_expiry: body.alpha_expiry,
+            beta_expiry: body.beta_expiry,
+            identities: SwapRequestIdentities::from_swap_request_body_identities(
+                body.identities,
+                id,
+                secret_source,
+            )?,
+            bob_socket_address: body.peer,
+        })
+    }
+}
+
+pub fn handle_post_swap<A: AliceSpawner>(
```

Comment:
:upside_down_face: Since you're reorganising stuff, maybe this could go at the top, since it's the focal point of the file. Same goes for the other handlers.